src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -84,17 +85,19 @@ public class BlockTree : IBlockTree
             IDb blockDb,
             IDb headerDb,
             IDb blockInfoDb,
+            IBlockInfoRepository blockInfoRepository,
             ISpecProvider specProvider,
             ITxPool txPool,
             ILogManager logManager)
-            : this(blockDb, headerDb, blockInfoDb, specProvider, txPool, new SyncConfig(), logManager)
+            : this(blockDb, headerDb, blockInfoDb, blockInfoRepository, specProvider, txPool, new SyncConfig(), logManager)
```

Comment:
wby both db and repo?

Reply:
Repository is used only for ChainLevelInfo, BlockTree uses blockInfoDb also for additional things

Reply:
as above, we need to review the locking logic